### PR TITLE
FSM arrow keys don't make the avatar walk

### PR DESCRIPTION
**Pull request by sethalves**
- in fullscreen-mirror mode, arrow keys move the camera but not the avatar

### CODE REVIEW

**Comment by sethalves** [comment]
https://highfidelity.fogbugz.com/f/cases/1138/In-mirror-mode-arrow-keys-should-only-move-camera-not-avatar

**Comment by sethalves** [comment]
to test:
- experiment with various camera modes and make sure they all act okay
- in FSM mode, observe that the arrow keys move the camera and wasd keys move the avatar

**Comment by hifi-gustavo** [comment]
Build console output for commit: 40a5e4abc892cb3ef0357cdc0b492a01327e40a4 available at: https://gist.github.com/bb13929afbd71c85e0d36acf7931a230<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-40a5e4a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-40a5e4a.dmg

**Comment by sethalves** [comment]
test this please

File: interface/resources/controllers/keyboardMouse.json [hunk 1]
```diff
@@ -63,6 +63,19 @@
...
+        { "from": { "makeAxis" : [
+                ["Keyboard.A", "Keyboard.Left", "Keyboard.TouchpadLeft"],
+                ["Keyboard.D", "Keyboard.Right", "Keyboard.TouchpadRight"]
+            ]
+          },
+          "when": "Application.CameraThirdPerson",
+          "to": "Actions.Yaw"
+        },
+        { "from": { "makeAxis" : [ ["Keyboard.A"], ["Keyboard.D"] ] },
+          "when": "Application.CameraFSM",
```
Comment (ZappoMan) [inline interface/resources/controllers/keyboardMouse.json]: wait, you want this? or you don't want this? I though in FSM you don't want to drive the yaw

**Comment by ZappoMan** [comment]
https://s-media-cache-ak0.pinimg.com/236x/61/e2/fe/61e2fe74aeca5053981e67b1087affa1.jpg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 40a5e4abc892cb3ef0357cdc0b492a01327e40a4 available at: https://gist.github.com/bb13929afbd71c85e0d36acf7931a230<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-40a5e4a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-40a5e4a.dmg

File: interface/resources/controllers/keyboardMouse.json (hunk 1 above)
Comment (sethalves) [inline interface/resources/controllers/keyboardMouse.json]: this one is for the A and D keys, which I want to still move the avatar.

**Comment by Nex-Pro** [comment]
@sethalves 

For a bucket full of flickery fun try  w+d+left arrow with 10 feet between the avatar and viewpoint.

Entity mode is awesome !

Strange rendering artifact on the avatar's face in mini mirror. A small version of the avatar is rendered on the face.


mirrorissue

**Comment by sethalves** [comment]
I'm pretty sure that reflection on the face isn't from this PR.  could you try that in master?

**Comment by Nex-Pro** [comment]
@sethalves checking now.

**Comment by Nex-Pro** [comment]
@sethalves  Worse in 5215 !

**Comment by Nex-Pro** [comment]
@sethalves I can rule out a graphics card issue as I changed the graphics card since posting the earlier observations.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 84ce29eb904c79e0827208ebf5b254c8700daf8c available at: https://gist.github.com/bb13929afbd71c85e0d36acf7931a230<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-84ce29e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-84ce29e.dmg

**Comment by sethalves** [comment]
yeah -- I've reproduced the reflection-on-face thing with that avatar.  it's unrelated to this PR.

Do you consider the jittering (when you hold w and d and an arrow key) unacceptable?

**Comment by sethalves** [comment]
test this please

**Comment by Nex-Pro** [comment]
yes the jittering is unacceptable  

maybe it's like the old movie camera issue whereby panning one way is good but the other is choppy

**Comment by Nex-Pro** [comment]
@sethalves  It looks like it's a shader issue due to the PBR materials on the scanned avatars

**Comment by Nex-Pro** [comment]
@sethalves  The cure for the jitter is  Install Gtx 1070

**Comment by hifi-gustavo** [comment]
Build console output for commit: 84ce29eb904c79e0827208ebf5b254c8700daf8c available at: https://gist.github.com/bb13929afbd71c85e0d36acf7931a230<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-84ce29e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-84ce29e.dmg

**Comment by Nex-Pro** [comment]
@sethalves  Do you require a re-test of this ?

**Comment by sethalves** [comment]
Well, I'm torn.  Probably what we should do is pass this and open another bug about the camera jitter.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 38314b180b20d350354edbe5bc5be8ca5897e495 available at: https://gist.github.com/bb13929afbd71c85e0d36acf7931a230<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-38314b1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8357-38314b1.dmg

**Pull request merged**